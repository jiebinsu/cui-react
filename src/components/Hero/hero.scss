@import "@coopdigital/shared-component--hero/dist/hero.css";

.coop-c-hero__outer {
  position: relative;
}
