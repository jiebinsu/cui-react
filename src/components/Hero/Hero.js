import React from "react";
import PropTypes from "prop-types";
import classNames from "../../utils/classNames";
import "./hero.scss";

const Hero = ({
  title,
  content,
  squircleBg,
  pullContent,
  ctaComponent,
  mediaComponent,
  children,
}) => {
  const rootClasses = classNames("coop-c-hero", [
    squircleBg && "coop-c-hero--squircle",
    mediaComponent && "coop-c-hero--media",
    pullContent && "coop-c-hero--pull",
  ]);

  return (
    <>
      <div className={rootClasses}>
        <div className="coop-c-hero__inner">
          <h1 className="coop-c-hero__title">{title}</h1>
          {mediaComponent && (
            <div className="coop-c-hero__media">{mediaComponent}</div>
          )}
          <div className="coop-c-hero__body">
            <p>{content}</p>
            {ctaComponent && (
              <div className="coop-c-hero__link">{ctaComponent}</div>
            )}
          </div>

          {squircleBg && (
            <div className="coop-c-hero__squircle" aria-hidden="true">
              <div className="coop-c-hero__squircle__inner">
                <svg
                  className="coop-c-hero__squircle__svg"
                  viewBox="0 0 1802 1726"
                >
                  <path
                    className="coop-c-hero__squircle__path"
                    d="M1743 446.3c-49.7-167.6-173.7-299.6-339.6-362C1079.6-28.2 724.1-28.2 397 84.9 232.6 146.7 109.2 278.6 58.5 446.8 19.7 581.6 0 721.4 0 862.8s19.7 281.2 58.5 416.5c50.8 167.6 174.2 299.6 339.6 362 162.2 56.2 331.5 84.7 502.9 84.7 172 0 341.3-28.4 504-85.2 164.9-62.4 288.3-193.8 338-362 39.4-135.8 59-275.6 59-416s-19.7-280.6-59-416.5z"
                  />
                </svg>
              </div>
            </div>
          )}
        </div>
      </div>
      <div className="coop-c-hero__outer">{children}</div>
    </>
  );
};

Hero.defaultProps = {
  squircleBg: false,
  pullContent: false,
  ctaComponent: null,
  mediaComponent: null,
  children: null,
};

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  squircleBg: PropTypes.bool,
  pullContent: PropTypes.bool,
  ctaComponent: PropTypes.node,
  mediaComponent: PropTypes.node,
  children: PropTypes.node,
};

export default Hero;
