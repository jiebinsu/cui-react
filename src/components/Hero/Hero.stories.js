import React from "react";
import Hero from "./Hero";
import ResponsiveImage from "../ResponsiveImage";

export default {
  title: "Base/Hero",
  component: Hero,
};

const heroTitle = "Supporting communities";
const content = `Find out what community and the power of local people means to Co-op. 
You'll discover the great things we can achieve when we work together to create a better society.`;

export const Basic = (args) => <Hero {...args} />;
Basic.args = {
  title: heroTitle,
  content,
  squircleBg: true,
};

const LinkButton = () => (
  <a href="http://google.com">More about how Co-op is supporting communities</a>
);
export const WithLinkCTA = () => (
  <Hero
    title={heroTitle}
    content={content}
    ctaComponent={<LinkButton />}
    squircleBg
  />
);

const Button = () => (
  <a href="https://coop.co.uk" className="coop-btn coop-btn--primary">
    Find out how you can help
  </a>
);
export const WithButtonCTA = () => (
  <Hero
    title={heroTitle}
    content={content}
    ctaComponent={<Button />}
    squircleBg
  />
);

const MediaComponent = () => {
  const src =
    "https://images.ctfassets.net/bffxiku554r1/7XDwxNP8ioupeMdJZxNS8/d23d2b802d6be0ecbcb7abd1ff33b159/coop-supporting-local-communities.png?fm=png&q=60&w=600&h=338";
  const alt = "How Co-op is supporting local communities";
  return <ResponsiveImage src={src} alt={alt} />;
};
export const WithMedia = () => (
  <Hero
    title={heroTitle}
    content={content}
    mediaComponent={<MediaComponent />}
    squircleBg
  />
);

export const WithMediaAndCTA = () => (
  <Hero
    title={heroTitle}
    content={content}
    ctaComponent={<Button />}
    mediaComponent={<MediaComponent />}
    squircleBg
  />
);
