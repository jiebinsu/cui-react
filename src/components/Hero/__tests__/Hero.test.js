import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";
import Hero from "../index";

describe("Hero", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Hero title="Title" content="Content description" />);
    expect(() => wrapper.unmount()).not.toThrow();

    const tree = renderer
      .create(<Hero title="Title" content="Content description" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow squircleBg attribute", () => {
    const tree = renderer
      .create(<Hero title="title" content="content" squircleBg />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow children and allow pullContent attribute", () => {
    const tree = renderer
      .create(
        <Hero title="title" content="content" pullContent>
          test
        </Hero>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow cta and media components", () => {
    const tree = renderer
      .create(
        <Hero
          title="title"
          content="content"
          pullContent
          ctaComponent={<button>button</button>}
          mediaComponent={<div>imageplaceholder</div>}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
