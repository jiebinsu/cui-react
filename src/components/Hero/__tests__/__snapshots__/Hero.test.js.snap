// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero should allow children and allow pullContent attribute 1`] = `
Array [
  <div
    className="coop-c-hero coop-c-hero--pull"
  >
    <div
      className="coop-c-hero__inner"
    >
      <h1
        className="coop-c-hero__title"
      >
        title
      </h1>
      <div
        className="coop-c-hero__body"
      >
        <p>
          content
        </p>
      </div>
    </div>
  </div>,
  <div
    className="coop-c-hero__outer"
  >
    test
  </div>,
]
`;

exports[`Hero should allow cta and media components 1`] = `
Array [
  <div
    className="coop-c-hero coop-c-hero--media coop-c-hero--pull"
  >
    <div
      className="coop-c-hero__inner"
    >
      <h1
        className="coop-c-hero__title"
      >
        title
      </h1>
      <div
        className="coop-c-hero__media"
      >
        <div>
          imageplaceholder
        </div>
      </div>
      <div
        className="coop-c-hero__body"
      >
        <p>
          content
        </p>
        <div
          className="coop-c-hero__link"
        >
          <button>
            button
          </button>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="coop-c-hero__outer"
  />,
]
`;

exports[`Hero should allow squircleBg attribute 1`] = `
Array [
  <div
    className="coop-c-hero coop-c-hero--squircle"
  >
    <div
      className="coop-c-hero__inner"
    >
      <h1
        className="coop-c-hero__title"
      >
        title
      </h1>
      <div
        className="coop-c-hero__body"
      >
        <p>
          content
        </p>
      </div>
      <div
        aria-hidden="true"
        className="coop-c-hero__squircle"
      >
        <div
          className="coop-c-hero__squircle__inner"
        >
          <svg
            className="coop-c-hero__squircle__svg"
            viewBox="0 0 1802 1726"
          >
            <path
              className="coop-c-hero__squircle__path"
              d="M1743 446.3c-49.7-167.6-173.7-299.6-339.6-362C1079.6-28.2 724.1-28.2 397 84.9 232.6 146.7 109.2 278.6 58.5 446.8 19.7 581.6 0 721.4 0 862.8s19.7 281.2 58.5 416.5c50.8 167.6 174.2 299.6 339.6 362 162.2 56.2 331.5 84.7 502.9 84.7 172 0 341.3-28.4 504-85.2 164.9-62.4 288.3-193.8 338-362 39.4-135.8 59-275.6 59-416s-19.7-280.6-59-416.5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="coop-c-hero__outer"
  />,
]
`;

exports[`Hero should render correctly 1`] = `
Array [
  <div
    className="coop-c-hero"
  >
    <div
      className="coop-c-hero__inner"
    >
      <h1
        className="coop-c-hero__title"
      >
        Title
      </h1>
      <div
        className="coop-c-hero__body"
      >
        <p>
          Content description
        </p>
      </div>
    </div>
  </div>,
  <div
    className="coop-c-hero__outer"
  />,
]
`;
